ComSemApp/student/views.py

```diff
@@ -799,17 +793,23 @@ def get_context_data(self, **kwargs) -> dict[str, dict[str, str | int | float]]:
         """
         context : dict[str, Any] = super(SpeakingPracticeResultsView, self).get_context_data(**kwargs)
         context['attempts'] : list[dict[str, str | float]] = {}
-        try:
-            attempt_ids : list[str] = dict(self.request.GET)['attempt_ids']
-        except:
-            return context
-        attempts : QuerySet[SpeakingPracticeAttempt] = SpeakingPracticeAttempt.objects.filter(student=self.student, pk__in=attempt_ids)
+        attempt_ids : list[str] = dict(self.request.GET).get('attempt', None)
+        session_ids : list[str] = dict(self.request.GET).get('session', None)
+
+        # if both are null
+        if(not (attempt_ids or session_ids)):
+            raise BadRequest("No attempt id or session id given")
+        
+        attempts : QuerySet[SpeakingPracticeAttempt] = SpeakingPracticeAttempt.objects.filter(
+            Q(session__student=self.student)
+            & Q(worksheet__course=self.course)
+            & (Q(id__in=attempt_ids) | Q(session__id__in=attempt_ids)))
```

Comment:
I am curious how this works. I understand that we will either have attempt_ids or session_id passed in. Does this filter just return whichever one actually gives back results. 
Would it not be more clear to check if attempt_ids is none and then run a query to get attempts from the session_id (or from the attempt_ids if  attempt_ids is not None)?

Reply:
This was intended to get results for both if both are given. The filter gets all attempts that are specifically given _and_ all attempts that are part of a given session.
There is a mistake here that I'm fixing now, and I'll add a doc comment.